### PR TITLE
WIP: Support for java 9 modules

### PR DESCRIPTION
**Pull request by weisJ**
This PR explores the possibility to support modules introduced with Java 9.
Current state is:
- the project can be compiled for Java 8 and
- the module-info.java is compiled for Java 9 giving support for people using modules.
- running requires adding necessary `--add-exports` as JVM arguments.

Note: The project needs to be build using Java 11 or higher.

What still needs to be done:
- Reducing the use of internal classes (though not easy. Some packages have been eliminated but only by mirroring the used code from the internal classes)
- Eliminating all illegal reflective access operations.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -94,19 +101,38 @@ allprojects {
 
     plugins.withType<JavaPlugin> {
         configure<JavaPluginExtension> {
-            sourceCompatibility = JavaVersion.VERSION_1_8
-            targetCompatibility = JavaVersion.VERSION_1_8
             withSourcesJar()
-            if (!skipJavadoc) {
+            if (!skipJavadoc && false) {
```
Comment (vlsi) [inline build.gradle.kts]: You can add `skipJavadoc=true` to your `$HOME/.gradle/gradle.properties` by the way.

**Comment by vlsi** [comment]
I see a lot of code seems to be borrowed from OpenJDK.
Should it be removed?

For instance: `BasicListUI#installKeyboardActions`, `BasicListUI#getInputMap`, `DarkTextUI#installKeyboardActions`, `DarkTextUI#getInputMap`, `DarkTextUI#getActionMap` , `DarkTextUI#updateFocusAcceleratorBinding` and so on.

The removal of the methods would automatically fix `access to internal classes`.

**Comment by weisJ** [comment]
The big problem here is that `BasicLookAndFeel` is super not extensible. There are horribly many places where methods and fields are `package private` or `private`. Things that should be accessible to any implementation that wants to extend basicLaf (which is sensible as it implements most of the behaviour and all other default laf do so too). It's true that most of the classes just copy the jdk implementation with the twist that everything was made `protected`. Most of the time if you want to change one method (which for example adds an ugly black border to a component) you'll have to reimplement most of the other methods that it uses too. It gets even worse if you want to access a variable that definitely should have gotten an accessor method (e.g. `isFileList` in `BasicListUI`). In this case you have no other choice than to copy most of the implementation.

The worst case are all the `package`private` or `private` event handlers in all the `Basic*UI` classes. You can't extend them and reimplementing them needs basically the whole ui implementation.

**Comment by weisJ** [comment]
After doing some more analysis, most of the places where `*.sun.*` packages are used can be replaced or at least reimplemented in a feasible manner. Though this can’t actually be achieved using Java 8. Most of the alternatives api that are AWT and Swing related haven’t been introduced until Java 9 or even 11. If there is demand for creating a jigsaw compatible version of darklaf it either:
- needs to be a port (probably using java 11).
- the conundrum project needs to specify certain export flags when running the program.

**Pull request closed without merging**